File: app/src/main/java/com/github/overpass/gather/model/usecase/image/ImageSourceUseCase.java
package com.github.overpass.gather.model.usecase.image;

import androidx.lifecycle.LiveData;

import com.github.overpass.gather.auth.register.add.ImageSource;
import com.github.overpass.gather.model.commons.SingleLiveEvent;

public class ImageSourceUseCase {

    private final SingleLiveEvent<ImageSource> imageSourceData;

    public ImageSourceUseCase() {
        imageSourceData = new SingleLiveEvent<>();
    }

    public LiveData<ImageSource> getImageSourceData() {
        return imageSourceData;
    }

    public void onGallery() {
        imageSourceData.setValue(ImageSource.GALLERY);
    }

    public void onCamera() {
        imageSourceData.setValue(ImageSource.CAMERA);
    }
}

File: app/src/main/java/com/github/overpass/gather/model/usecase/image/ImageSourceUseCase.kt
package com.github.overpass.gather.model.usecase.image

import androidx.lifecycle.LiveData
import com.github.overpass.gather.model.commons.SingleLiveEvent
import com.github.overpass.gather.screen.auth.register.add.ImageSource

class ImageSourceUseCase constructor(
        private val imageSourceData: SingleLiveEvent<ImageSource> = SingleLiveEvent()
) {

    fun getImageSourceData(): LiveData<ImageSource> {
        return imageSourceData
    }

    fun onGallery() {
        imageSourceData.value = ImageSource.GALLERY
    }

    fun onCamera() {
        imageSourceData.value = ImageSource.CAMERA
    }
}
